featuretools/computational_backends/feature_set_calculator.py

```diff
@@ -337,6 +337,9 @@ def _add_ancestor_relationship_variables(self, child_df, parent_df,
                             left_on=relationship.child_variable.id,
                             right_on=relationship.child_variable.id)
 
+        # ensure index is maintained
+        df = df.set_index(relationship.child_entity.index, drop=False)
```

Comment:
Probably not a big deal, but `inplace` looks like it might be faster.
```
In [1]: import pandas as pd

In [2]: df10k = pd.DataFrame({'a': range(10000)}, index=range(10000))

In [3]: df100k = pd.DataFrame({'a': range(100000)}, index=range(100000))

In [4]: %timeit df10k.set_index('a', drop=False)
312 µs ± 7.26 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [5]: %timeit df100k.set_index('a', drop=False)
912 µs ± 50.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [6]: %timeit df10k.set_index('a', drop=False, inplace=True)
101 µs ± 5.42 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [7]: %timeit df100k.set_index('a', drop=False, inplace=True)
113 µs ± 25.7 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```